### PR TITLE
build: update dependences

### PR DESCRIPTION
**Pull request by lklimek**
## Issue being fixed or feature implemented

Updated dependencies

## What was done?

* updated multiple dependencies
* removed atty dependency which is not needed anymore, as the feature is already in rust std lib
* updated source code to reflect dependency changes, mainly in `derive_more` 

Note: requires #2070 to be merged first

## How Has This Been Tested?

GHA

## Breaking Changes

None

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added or updated relevant unit/integration/functional/e2e tests
- [ ] I have added "!" to the title and described breaking changes in the corresponding section if my code contains any
- [ ] I have made corresponding changes to the documentation if needed

**For repository code-owners and collaborators only**
- [x] I have assigned this pull request to a milestone

## Release Notes

- **New Features**
	- Retained and clarified documentation comments in relevant methods for better understanding.

- **Bug Fixes**
	- Various dependency updates may include bug fixes and performance improvements.

- **Documentation**
	- Updated the version of the `protoc - protobuf compiler` in the README.

- **Refactor**
	- Simplified display formatting in several enums and structs for cleaner code aesthetics without changing functionality.

- **Chores**
	- Updated dependency versions across multiple packages to enhance stability and performance.

### CODE REVIEW

**Comment by shumkov** [comment]
It seems the branch is outdated since I see changes from other PRs

File: packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_purchase_transition/v0/mod.rs [hunk 1]
```diff
@@ -19,7 +19,7 @@ pub use super::super::document_base_transition::IDENTIFIER_FIELDS;
     derive(Serialize, Deserialize),
     serde(rename_all = "camelCase")
 )]
-#[display(fmt = "Base: {}", "base")]
+#[display( "Base: {}", "base")]
```
[the remark this answers is not in the document]
Comment (lklimek) [reply packages/rs-dpp/src/state_transition/state_transitions/document/documents_batch_transition/document_transition/document_purchase_transition/v0/mod.rs]: Only derive_more 1.0 is in use

File: Dockerfile [hunk 2]
```diff
@@ -30,7 +30,7 @@
 # SCCACHE_SERVER_PORT port to avoid conflicts in case of parallel compilation
 
 ARG ALPINE_VERSION=3.18
-ARG PROTOC_VERSION=25.2
+ARG PROTOC_VERSION=27.3
```
Comment (shumkov) [inline Dockerfile:33]: Please update readme and dev container

**Review by shumkov — CHANGES_REQUESTED** [review]

**Review by shumkov — APPROVED** [review]

**Review by QuantumExplorer — APPROVED** [review]

File: Dockerfile (hunk 2 above)
Comment (lklimek) [reply Dockerfile:33]: done

**Review by QuantumExplorer — APPROVED** [review]

**Review by shumkov — APPROVED** [review]

**Pull request merged**